index.js

```diff
@@ -180,7 +187,7 @@ function addRouteRateHook (pluginComponent, params, routeOptions) {
 
 function rateLimitRequestHandler (pluginComponent, params) {
   const { rateLimitRan, store } = pluginComponent
-  const timeWindowString = ms.format(params.timeWindow, true)
+  const timeWindowString = ms.format(typeof params.timeWindow === 'function' ? params.timeWindow() : params.timeWindow, true)
```

Comment:
Minor feedback:

Passing no arguments is not a valid signature according to the type information I think, you can make the parameters optional

Also, just for consistency, let's `await` this as well since it can return a Promise

```suggestion
  const timeWindowString = ms.format(typeof params.timeWindow === 'function' ? await params.timeWindow() : params.timeWindow, true)
```

Reply:
That was not optimal. Thanks for spotting. I simplified it a bit. No need for optional parameters. Should be better now.

Reply:
Yep, looks great!

Thanks